src/core/src/package_managers/AptitudePackageManager.py

```
@@ -359,6 +367,73 @@ def get_package_size(self, output):
+        try:
+            # todo: adding space between the patch_mode and value since, we will have to do that if we have to add a patch_mode that did not exist before
+            self.composite_logger.log("Updating default system settings for auto OS updates. [Patch Mode={0}. Value={1}]".format(str(patch_mode), value))
+            image_default_patch_mode = self.env_layer.file_system.read_with_retry(self.image_default_patch_mode_file_path)
+            patch_mode_to_update = patch_mode + ' "' + value + '";'
+            patch_mode_found_in_file = False
+            updated_patch_mode = ""
+            settings = image_default_patch_mode.strip().split('\n')
+            for i in range(len(settings)):
+                # ToDo: do we need a check for value as well, to check if the value is not the same as the one sent in to replace it, as that will be an unnecessary write
+                if patch_mode in settings[i]:
+                    settings[i] = patch_mode_to_update
+                    patch_mode_found_in_file = True
+                updated_patch_mode += settings[i] + "\n"
+            if not patch_mode_found_in_file:
+                updated_patch_mode += patch_mode_to_update + "\n"
+            #ToDo: This adds some whitespace at the beginning of the first line in the settings file which is auto adjusted in the file later, so shouldn't have any issues right now. strip()/lstrip() on the string, does not work, will have to test accross versions and identify teh impact
+            self.env_layer.file_system.write_with_retry(self.image_default_patch_mode_file_path, '{0}'.format(updated_patch_mode.lstrip()), mode='w+')
+        except Exception as error:
+            self.composite_logger.log_error("Error occurred while updating default system settings for auto OS updates. [Patch Mode={0}] [Error={1}]".format(str(patch_mode), repr(error)))
```

"Error occurred while updating default system settings for auto OS updates"
- what does 'default' signify here? It doesn't seem to qualify system settings any further than the only interpretation possible. As I wrote this, it dawned on me that the code is only set up to do that initial disable. Since we know configuration has to come pretty much soon after this, look further ahead if possible now.

---

The word 'default' is misleading here. The function is meant to update any system setting with a given value, which was with the future changes in mind. So the function only writes one setting in the file for every invocation and since the value is passed as an arg, it can be enabled or disabled. Let me know if this answers your concern here, or if there is something I'm missing